src/main/kotlin/com/fortysevendeg/hood/BenchmarkFilter.kt

```diff
@@ -0,0 +1,21 @@
+package com.fortysevendeg.hood
+
+import arrow.core.Option
+import com.fortysevendeg.hood.models.Benchmark
+
+private fun List<Benchmark>.filterUsingRegex(
+  maybeRegex: Option<Regex>,
+  rFilter: ((Benchmark) -> (Boolean)) -> List<Benchmark>
+): List<Benchmark> =
+  maybeRegex.fold({ this }) { regex -> rFilter { bm: Benchmark -> regex.matches(bm.getKey()) } }
+
+fun List<Benchmark>.filterBenchmark(includeRegex: Option<Regex>): List<Benchmark> =
+  filterUsingRegex(includeRegex) { this.filter(it) }
```

Comment:
I am not sure about it, just to know, but in `swift` we could do something like this:
```kotlin
filterUsingRegex(includeRegex, filter)
filterUsingRegex(includeRegex, filterNot)
```

Reply:
Is the same but a little bit more explicit due to kotlin limitations